sycl/doc/KernelProgramCache.md

```diff
@@ -214,6 +254,26 @@ name, device as well as any specialization constants. These get added to
 `BuildResult` and are cached. The `BuildResult` structure is specialized with
 either `PiKernel` or `PiProgram`<sup>[1](#remove-pointer)</sup>.
 
+### Inter-process safety
+For on-disc cache there might be access collisions for accessing the same file
+from different instance of SYCL applications:
+- write collision happens when 2 instances of the same application are started
+to write to the same cache file;
+- read collision may happen if one application is writing to the file and the
+other instance of the application is trying to read from it while write
+operation is not finished.
+To avoid collisions the file in on-disc cache are locked for read-write access
+until write operation is finished.
```

Comment:
Should the process lock the whole cache directory explicitly? For example, in the same way as `apt-get` does it.

Reply:
I will learn what is pros and cons for locking whole directory and provide an update in next revision of the doc.